platform/measurement_utils.cpp

```diff
@@ -53,6 +74,43 @@ std::string ToStringPrecisionLocale(platform::Locale const & loc, double d, int
   return out;
 }
 
+std::string ToStringPrecisionLocale_new(platform::Locale const & loc, double d, int pr)
+{
+  #define BUF_SIZE 50
+
+  char formatBuf[BUF_SIZE];
+  switch(pr)
+  {
+    case 0: strcpy(formatBuf, "%0.0f"); break;
+    case 1: strcpy(formatBuf, "%0.1f"); break;
+    default: std::snprintf(formatBuf, BUF_SIZE, "%%0.%df", pr); break;
+  }
```

Comment:
If we're touching the performance... What about this?

```suggestion
  std::string formatBuf("%0.0f");
  if (pr > 0)
  {
    if (pr < 10)
      formatBuf[3] = '0' + pr;  // E.g. replace %0.0f with %0.1f
    else
      LOG(LERROR, ("Too big unsupported precision", pr));
  }
```